panther_analysis_tool/main.py

```diff
@@ -118,6 +123,17 @@
     "Tags",
 ]
 
+# Available destination names for use by rules in tests
+AVAILABLE_DESTINATION_NAMES = ("ExampleDestinationName",)
+
+# The rule executor needs to map any destination names returned by the `destination` function
+# to actual destination objects. We are using a fake Destination dataclass, since we
+# just need the name and identifier fields.
+OUTPUT_DISPLAY_NAMES = {
+    name: FakeDestination(destination_id=str(uuid4()), destination_display_name=name)
+    for name in AVAILABLE_DESTINATION_NAMES
+}
```

Comment:
So in all cases, it appears the only destination that panther detection unit tests can return is `ExampleDestinationName`?

In general, detection unit tests should be able to return any string since we don't know what users may have named their detections. Is their a way to relax this requirement? Or at least we should have some configurability for end users on this list of destination names.

Reply:
That's a very good observation. Maybe we should add a command-line parameter for now to allow users to define a list of available names. Or perhaps ignore destinations altogether for the time being? I remember you mentioned before that due to their dynamic nature, it might be a bit difficult to handle them at the moment.

Reply:
In my opinion the ideal behavior for now is to just verify that a string is returned, which is the current behavior (prior to this change).

If that's not possible, ignoring it might be best. The problem is that if we put this in, even with giving users the option to configure it, it will fail CI/CD build jobs for any users that are using the custom destinations feature until they configure it.

If it's not too much work, can we add the option to configure a list of destination names (either as a command line flag or in a configuration file like `destinations.txt` or something), and if none are configured then skip the destination checks altogether? That way we can tell users to start migrating towards defining destinations, without a hard cutover that will immediately break CI pipelines (since most customers pull the latest version of `panther_analysis_tool` in their pipelines).

Reply:
@nhakmiller I added an `--available-destination <NAME>` parameter that can be repeated in order to provide for multiple values. I removed the default, constant destination name and instead tried to match that `destinations` returns a list of strings by inspecting the corresponding exception message. The check is far from perfect, but I thought it could suffice for now. I didn't want to modify the rule logic at the moment, but if we have to we can do that too.

Note that only when a user provides an `--available-destination` parameter, is strict checking for destination names activated, in which case they'll start to get failures if names do not match. 

Let me know if this approach could work.

Reply:
Yeah that's perfect, I'll re-review & probably approve today.